com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/HDTarget.cs

```diff
@@ -59,15 +59,10 @@ sealed class HDTarget : Target, IHasMetadata, ILegacyTarget
         [SerializeField]
         string m_CustomEditorGUI;
 
-        static NodeTypeCollection s_HDNodeTypes = NodeTypes.AllBuiltin + typeof(HDSceneColorNode)
-                                                                       + typeof(DiffusionProfileNode)
-                                                                       + typeof(ExposureNode)
-                                                                       + typeof(EmissionNode)
-                                                                       + typeof(ParallaxOcclusionMappingNode);
-
         public override bool IsNodeAllowedByTarget(Type nodeType)
         {
-            return s_HDNodeTypes.Contains(nodeType);
+            SRPFilterAttribute srpFilter = NodeClassCache.GetAttributeOnNodeType<SRPFilterAttribute>(nodeType);
```

Comment:
Just a question. is this method here IsNodeAllowedByTarget. Allow us (Later) to filter a node also per master node?

so for example in HDLitMaster node, I could reimplement this mode to reject some specific node that aren't compatible?

Side note: All the current lighting node (those that are use to access fog or light color in Universal aren't supported by HDRP, be sure they are filtered).

Reply:
Yes, extensibility is definitely a primary concern with the system. I thought about including it off the bat, but since no subtargets currently require it I figured it could be put off till later

Reply:
Possible other uses would be defining allowed nodes based on build target (like max ps version for XBox, etc)